File: official/docs/php/current/smartrate/retrieve-standalone-smartrate-deliver-on.php
<?php

$client = new \EasyPost\EasyPostClient('EASYPOST_API_KEY');

$params = [
    'from_zip' => '10001',
    'to_zip' => '10002',
    'desired_delivery_date' => '2024-07-18',
    'carriers' => ['UPS'],
];

$results = $client->smartRate->recommendShipDate($params);

echo $results;

File: official/docs/python/current/smartrate/retrieve-standalone-smartrate-deliver-on.py
import easypost

client = easypost.EasyPostClient("EASYPOST_API_KEY")

results = client.smartrate.recommend_ship_date(
    from_zip="10001",
    to_zip="10002",
    desired_delivery_date="2024-07-18",
    carriers=["usps"],
)

print(results)
